### ISSUE TITLE
Allow to display PDF manuals also for registers and maybe functions

### ISSUE THREAD

**Issue by utnubu1** [NationalSecurityAgency/ghidra#1654]
**Is your feature request related to a problem? Please describe.**
I am currently integrating an embedded processor with lots of special-function registers and a custom system calls  into Ghidra and is actually quite annoying to search the manual each time to find out what action is performed by an instruction modifying a SFR register. The same holds for the exact meaning of invoked system calls. 

**Describe the solution you'd like**
It would be quite cool if the manual  for register meanings and system calls could be linked in the same way as the Processor Manual for instructions currently is. I currently realized both with a custom Plugin which reads distinct idx files in the same manner as the Processor Manual code. If anyone else is interested in this i could port that plugin to the Ghidra core and submit a pull request.

**Describe alternatives you've considered**
If thats a Feature no one else wants to have i will just carry on with my own Plugin :)

**Comment by Marc-Bender** [comment]
yep just looked at one of my own binary blobs for an atmel avr and id definitely find that useful. 
more so i found that the elf file atmel studio produces even contains the name of the specific avr in use which would make things even more interresting . 

adding in all registers and all datasheets for all variants of all supported processors plus all that my come is a herculean effort however and thus likely not gonna happen. 
    --> looking at how long some issues are open here already tells a story of its own that supports that this feature is likely not gonna be realized

**Comment by utnubu1** [comment]
@Marc-Bender if you are still interested in this i can share the Plugin i use at the moment.

**Comment by Marc-Bender** [comment]
i rarely ever do reversing. when i do i mainly do on my own stuff and for curiosity only. so i really do not have a need for it in the real world but i can see why it is useful and should exist...
